client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractRequestTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client;
+
+import org.elasticsearch.cluster.ClusterModule;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContent;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+/**
+ * Base class for HLRC request parsing tests.
+ *
+ * This case class facilitates generating client side request test instances and
+ * verifies that they are correctly parsed into server side request instances.
+ *
+ * @param <C> The class representing the request on the client side.
+ * @param <S> The class representing the request on the server side.
+ */
+public abstract class AbstractRequestTestCase<C extends ToXContent, S> extends ESTestCase {
+
+    private static final int NUMBER_OF_TEST_RUNS = 20;
```

Comment:
should this be overridable?

Reply:
We can make it overridable. However all the other parsing tests don't do this either and have the number of tests runs also fixed to 20. So I think for now this is ok?

Reply:
Do we need this? Our tests run a bazillion times a day in CI. Is 20x a bazillion buying us anything?

Reply:
> Do we need this? Our tests run a bazillion times a day in CI. Is 20x a bazillion buying us anything?

Good question. We do this in the other xcontent parsing tests too. I personally like like this during development, so that silly mistakes are caught before I merge into master. On the other hand, I can also run the tests with `-Dtests.iters=N` flag. Like you say we keep paying the price until these tests are no longer run in CI. So from that perspective it makes sense to remove the loop. If we do this here then we should also do this in the other xcontent parsing tests. wdyt?

Reply:
++ im fine w/ removing. I develop using tests.iters quite typically on the hlrc unit tests.

Reply:
I've removed the iterations in this base class and `AbstractResponseTestCase`: e5b1ca778702f9f068dc2819d311412b6c4accf8

Reply:
>If we do this here then we should also do this in the other xcontent parsing tests. wdyt?

+1